src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/food/Juice.java

```diff
@@ -65,39 +61,12 @@ public ItemConsumptionHandler getItemHandler() {
              * Minecraft has been broken when it comes to Saturation potions for a long time
              */
             for (PotionEffect effect : effects) {
-                if (effect.getType().equals(PotionEffectType.SATURATION)) {
-                    p.addPotionEffect(effect);
-                    break;
-                }
+                p.addPotionEffect(effect);
+                break;
```

Comment:
Why change this? You're now always going to just re-add the first effect which could be anything

Reply:
Because setting the item of the event to ``null`` basically makes it so that the player is eating air, and so no vanilla effects will be applied from the original potion. This ensures that the effects do get applied in the first place.

Reply:
What does the setting the item to null have to do with this change?
You have just changed the behavior to "Add saturation effect if it was saturation" to "Add effect whatever that was".
This does not check the item of the event at all. Since all juices only have saturation you don't even see the difference, if you reverted this to the original everything else you changed would still work fine so I am confused why you even changed it in the first place.

Reply:
Actually, golden apple juice gives absorption. It is because of golden apple juice that I added this.

Reply:
Yes but this was only to fix absorption. Minecraft applies all effects from potions just fine EXCEPT it forgets about saturation, so we add it once more. That's all commented right above the spot you edited:
"/*
 * Fix for Saturation on potions is no longer working,
 * Minecraft has been broken when it comes to Saturation potions for a long time
 */"

By doing this, you are reapplying all of the effects - which changes nothing since you are applying saturation too which was what this fix addressed, but you are applying for example absorption twice when it should only be applied once. It's redundant

Reply:
Setting the item to null will not apply absorption